src/ILVerify/src/InstantiatedGenericParameter.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Internal.TypeSystem
+{
+    internal sealed partial class InstantiatedGenericParameter : GenericParameterDesc
+    {
+        private readonly GenericParameterDesc _genericParam;
+        private Instantiation _typeInstantiation;
+        private Instantiation _methodInstantiation;
+
+        public GenericParameterDesc GenericParameter
+        {
+            get
+            {
+                return _genericParam;
+            }
+        }
+
+        internal static Instantiation CreateGenericTypeInstantiaton(Instantiation instantiation)
+        {
+            if (instantiation.Length == 0)
+                return instantiation;
+
+            var genericInstantiation = CreateGenericInstantiation(ref instantiation);
+
+            foreach (var parameter in genericInstantiation)
+                ((InstantiatedGenericParameter)parameter)._typeInstantiation = genericInstantiation;
+
+            return genericInstantiation;
+        }
+
+        internal static Instantiation CreateGenericMethodInstantiation(Instantiation typeInstantiation, Instantiation methodInstantiation)
+        {
+            if (methodInstantiation.Length == 0)
+                return methodInstantiation;
+
+            var genericInstantiation = CreateGenericInstantiation(ref methodInstantiation);
+
+            foreach (var parameter in genericInstantiation)
+            {
+                var par = (InstantiatedGenericParameter)parameter;
+                par._typeInstantiation = typeInstantiation;
+                par._methodInstantiation = genericInstantiation;
+            }
+
+            return genericInstantiation;
+        }
+
+        private static Instantiation CreateGenericInstantiation(ref Instantiation from)
```

Comment:
Nit: Change the name to `fromInstantiation`. (from is keyword - the keyword highlighting in the github UI makes it look odd if nothing else)